### PR TITLE
[#4] Add a background color option

### PR DESCRIPTION
**Pull request by ppuleo**
- Add a background color option
- Document the existing options
- Update the demo to use all options

### CODE REVIEW

**Comment by ppuleo** [comment]
Hi @stewartlord, here's an idea for allowing the background color to be set as an option. If other options are to be exposed, maybe an options object would make sense as the constructor argument instead of a list? I'm happy to do that if you prefer.

**Comment by stewartlord** [comment]
Hey Phil, so sorry for the delay. I just had a baby so I'm a bit overwhelmed at the moment. I like the idea of exposing more options as a hash if you are still willing to do that. We could keep hash as a standard argument since it is the only required param.

**Comment by ppuleo** [comment]
Hi @stewartlord, no problem and congratulations! I'll update the PR with the options hash approach.

**Comment by stewartlord** [comment]
Hi Phil, just cleaning up and I came across this PR. I was wondering if you wanted to keep it open?

**Comment by ppuleo** [comment]
Hi @stewartlord, sorry I meant to update this and resubmit. I'll revisit it this morning.

**Comment by ppuleo** [comment]
Hi @stewartlord, I've updated the PR with the changes we discussed plus a couple of other changes I've been using:
1. Updated the margin calculation to ensure the cells are always centered in the image.
2. If no hash is specified, create a reasonably randomized hash from the current time.
3. Optionally accept the size and margin arguments instead of the options object argument so the previous constructor signature still works for backward compatibility. 

Thanks again!

**Comment by stewartlord** [comment]
Ooh nice work buddy 👍 Just checking the diffs now..

File: README.md [hunk 1]
```diff
@@ -9,18 +9,34 @@ Demo
...
+* **hash** - [Optional] A unicode string that will be used to generate the image. Defaults to a random hash based on the current time.
+* **options** - [Optional] An options object used to customize the generated image.
+    * **background** - The background color expressed as an rgba value array to use for the image background. For example, use [255,0,0,255] for red. Defaults to an opaque light gray [240,240,240,255].
+    * **margin** - The decimal fraction of the size to use for margin. For example, use 0.2 for a 20% margin. Defaults to 0.1 for a 10% margin.
+    * **size** - The size in pixels of the height and width of the generated (square) image. Defaults to 64 pixels.
+
 Usage
 -----
 ```js
+// set options
```
Comment (stewartlord) [inline README.md]: Would you be opposed to a basic usage example with just hash and inline size {size: 420} -- followed by the more complete usage?

File: demo.html [hunk 2]
```diff
@@ -1,25 +1,87 @@
...
+          <p>Specify just a hash, use default background color, margins, and size.</p>
+<pre>
+  var hash = 'd6fe8c82fb0abac17a702fd2a94eff37';
+  var data = new Identicon(hash).toString();
+  document.write('&lt;img src="data:image/png;base64,' + data + '"&gt;');</pre>
+            <script type="text/javascript">
+              hashes = [
+              {id: 'stewartlord', hash: 'd6fe8c82fb0abac17a702fd2a94eff37'},
+              {id: 'jasonlong',   hash: '46384036044a604b6b3316fc167fc15f'},
+              {id: 'infusion',    hash: '373ff2375be91ebc97501e67d24ccd4d'}
```
Comment (stewartlord) [inline demo.html]: Add your username??

**Comment by stewartlord** [comment]
Looks great! Just a couple of minor suggestions

File: README.md (hunk 1 above)
Comment (ppuleo) [inline README.md]: Sure, makes sense. That's probably the most common use case. I'll revise.

File: demo.html (hunk 2 above)
Comment (ppuleo) [inline demo.html]: ha, ok. :smile:

**Comment by ppuleo** [comment]
ok, updated the usage section with simple/advanced examples and added my username to the demo. Good suggestions.

File: identicon.js [hunk 3]
```diff
@@ -11,29 +11,45 @@
...
-    Identicon = function(hash, size, margin){
-        this.hash   = hash;
-        this.size   = size   || 64;
-        this.margin = margin || .08;
-    }
+    Identicon = function(hash, options){
+        this.defaults = {
+            background: [240, 240, 240, 255],
+            hash:       this.createHashFromString((new Date()).toISOString()),
+            margin:     0.10,
```
Comment (stewartlord) [inline identicon.js]: Sorry I missed this before. Can we keep the old default margin (.08)?

Comment (ppuleo) [inline identicon.js]: oh, right. I meant to change that back after testing. Will fix.

**Comment by ppuleo** [comment]
Restored the margin default to .08

File: README.md [hunk 4]
```diff
@@ -9,18 +9,48 @@ Demo
 ----
 [View Demo](https://rawgithub.com/stewartlord/identicon.js/master/demo.html)
 
+Options
+----
+* **hash** - [Optional] A unicode string that will be used to generate the image. Defaults to a random hash based on the current time.
+* **options** - [Optional] An options object used to customize the generated image.
+    * **background** - The background color expressed as an rgba value array to use for the image background. For example, use [255,0,0,255] for red. Defaults to an opaque light gray [240,240,240,255].
+    * **margin** - The decimal fraction of the size to use for margin. For example, use 0.2 for a 20% margin. Defaults to 0.1 for a 10% margin.
```
Comment (stewartlord) [inline README.md]: So close! Says 10% here :(

Comment (ppuleo) [inline README.md]: Crap, sorry about that. Fixed.

**Comment by stewartlord** [comment]
Thanks Phil!

**Comment by ppuleo** [comment]
Thanks @stewartlord, much appreciated!

**Pull request merged**